fgpv-vpgf/fgpv-vpgf
# feat(identify): improvements to identify flow

## Description
Specifies four different identify modes: details, highlight, silent and none. See #2595 for more details.
Setting identify mode to `none` can disable the identify functionality at runtime as described in #1985.

### How to test
1. load `index-samples` page
2. type `RZ.mapInstances[0].identifyMode` in the console // identify mode defaults to 'details'
3. click on things, see them get highlighted and displayed in the details panel
4. type the following in the console (this will remove all but one result from each sublayer)
```ts
RZ.mapInstances[0].identify.subscribe(identifyResults =>
	identifyResults.forEach(result => {
		if (result.data.length > 0) {
		        result.data.splice(1);
		}
	})
);
```
5. click on things, see only one feature get highlighted and displayed in the details panel
6. type `RZ.mapInstances[0].identifyMode = 'highlight';` in the console
7. click on things - they are only highlighted now (still only one feature because the filter is still working)
8. type `RZ.mapInstances[0].identifyMode = 'silent';` in the console
8. type the following in the console
```ts
RZ.mapInstances[0].identify.subscribe(identifyResults => console.log(identifyResults));
```
10. click on things - see nothing happen on the map, but the results are printed in the console still
11. type `RZ.mapInstances[0].identifyMode = 'none';` in the console
12. click on this - notice nothing is happening
13. eat a muffin

Closes #2595 
Relates #1985 

## Testing

:eyes: 

## Documentation

Inline comments.

## Checklist



Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/2597)
